ui/src/app/views/export-data-set-modal.tsx

```diff
@@ -69,17 +82,32 @@ class ExportDataSetModal extends React.Component<
   }
 
   async generateQuery() {
-    const {workspaceNamespace, workspaceFirecloudName} = this.props;
-    const dataSet: DataSetRequest = {
-      name: '',
-      conceptSetIds: this.props.dataSet.conceptSets.map(cs => cs.id),
-      cohortIds: this.props.dataSet.cohorts.map(c => c.id),
-      values: this.props.dataSet.values,
-      includesAllParticipants: this.props.dataSet.includesAllParticipants,
+    const {dataSet, workspaceNamespace, workspaceFirecloudName} = this.props;
+    const dataSetRequest: DataSetRequest = {
+      name: dataSet.name,
+      conceptSetIds: dataSet.conceptSets.map(cs => cs.id),
+      cohortIds: dataSet.cohorts.map(c => c.id),
+      values: dataSet.values,
+      includesAllParticipants: dataSet.includesAllParticipants,
     };
-    const sqlQueries = await dataSetApi().generateQuery(
-      workspaceNamespace, workspaceFirecloudName, dataSet);
-    this.setState({queries: sqlQueries.queryList});
+    dataSetApi().generateCode(
+      workspaceNamespace,
+      workspaceFirecloudName,
+      KernelTypeEnum.Python.toString(),
+      dataSetRequest).then(pythonCode => {
+        const queries = this.state.queries;
+        queries.set(KernelTypeEnum.Python, pythonCode.code);
+        this.setState({queries: queries});
```

Comment:
Is there a way to do this in one line?  "appendToState", "mergeState", etc?

Reply:
I don't know. I was asking Brett, but I'm not sure. I think Lodash might be able to do it? But couldn't figure out how.